protocols/gossipsub/src/config.rs

```diff
@@ -48,61 +50,288 @@ pub enum ValidationMode {
     /// is 12).
-    pub mesh_n_high: usize,
+    mesh_n_high: usize,
 
-    /// Number of peers to emit gossip to during a heartbeat (D_lazy in the spec, default is 6).
-    pub gossip_lazy: usize,
+    /// Affects how peers are selected when pruning a mesh due to over subscription.
+    //  At least `retain_scores` of the retained peers will be high-scoring, while the remainder are
+    //  chosen randomly (D_score in the spec, default is 4).
+    retain_scores: usize,
+
+    /// Minimum number of peers to emit gossip to during a heartbeat (D_lazy in the spec,
+    /// default is 6).
+    gossip_lazy: usize,
+
+    /// Affects how many peers we will emit gossip to at each heartbeat.
+    /// We will send gossip to `gossip_factor * (total number of non-mesh peers)`, or
+    /// `gossip_lazy`, whichever is greater. The default is 0.25.
+    gossip_factor: f64,
 
     /// Initial delay in each heartbeat (default is 5 seconds).
-    pub heartbeat_initial_delay: Duration,
+    heartbeat_initial_delay: Duration,
 
     /// Time between each heartbeat (default is 1 second).
-    pub heartbeat_interval: Duration,
+    heartbeat_interval: Duration,
 
     /// Time to live for fanout peers (default is 60 seconds).
-    pub fanout_ttl: Duration,
+    fanout_ttl: Duration,
 
-    /// The maximum byte size for each gossip (default is 2048 bytes).
-    pub max_transmit_size: usize,
+    /// The number of heartbeat ticks until we recheck the connection to explicit peers and
+    /// reconnecting if necessary (default 300).
+    check_explicit_peers_ticks: u64,
+
+    /// The maximum byte size for each gossip (default is 65536 bytes).
+    ///
+    /// This represents the maximum size of the entire protobuf payload. It must be at least
+    /// large enough to support basic control messages. If Peer eXchange is enabled, this
+    /// must be large enough to transmit the desired peer information on pruning. It must be at
+    /// least 100 bytes. Default is 65536 bytes.
+    max_transmit_size: usize,
 
     /// Duplicates are prevented by storing message id's of known messages in an LRU time cache.
     /// This settings sets the time period that messages are stored in the cache. Duplicates can be
     /// received if duplicate messages are sent at a time greater than this setting apart. The
     /// default is 1 minute.
-    pub duplicate_cache_time: Duration,
+    duplicate_cache_time: Duration,
+
+    /// When set to `true`, prevents automatic forwarding of all received messages. This setting
+    /// allows a user to validate the messages before propagating them to their peers. If set to
+    /// true, the user must manually call `validate_message()` on the behaviour to forward message
+    /// once validated (default is `false`). Furthermore, the application may optionally call
+    /// `invalidate_message()` on the behaviour to remove the message from the memcache. The
```

Comment:
I think this is outdated and now included in `report_message_validation_result`, right?

Reply:
Correct.